### PR TITLE
Migrate to pre-commit ci

### PR DESCRIPTION
**Pull request by Sparks29032**
Good to merge. Pre-commit CI is set up for this repo.

This removes the `pre-commit.yml` file as `pre-commit.ci` should automatically run for every pull request after installation.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,5 +1,14 @@
 default_language_version:
     python: python3
+ci:
+    autofix_commit_msg: |
+        [pre-commit.ci] auto fixes from pre-commit hooks
+    autofix_prs: true
+    autoupdate_branch: 'pre-commit-autoupdate'
+    autoupdate_commit_msg: '[pre-commit.ci] pre-commit autoupdate'
+    autoupdate_schedule: monthly
+    skip: [no-commit-to-branch]
```
Comment (Sparks29032) [inline .pre-commit-config.yaml:10]: This skips the ci from running this hook, but it still runs locally.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,5 +1,14 @@
 default_language_version:
     python: python3
+ci:
+    autofix_commit_msg: |
+        [pre-commit.ci] auto fixes from pre-commit hooks
+    autofix_prs: true
+    autoupdate_branch: 'pre-commit-autoupdate'
+    autoupdate_commit_msg: '[pre-commit.ci] pre-commit autoupdate'
+    autoupdate_schedule: monthly
```
Comment (Sparks29032) [inline .pre-commit-config.yaml:9]: Shortest time is weekly, but monthly seemed fine. Can also extend to quarterly.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -1,5 +1,14 @@
 default_language_version:
     python: python3
+ci:
+    autofix_commit_msg: |
+        [pre-commit.ci] auto fixes from pre-commit hooks
+    autofix_prs: true
+    autoupdate_branch: 'pre-commit-autoupdate'
```
Comment (Sparks29032) [inline .pre-commit-config.yaml:7]: Updates on a branch named `pre-commit-autoupdate` so it does not change our main branch.

**Comment by sbillinge** [comment]
I linked diffpy org at pre-commit.ci.  I wasn't quite sure what else I needed to do?

**Comment by Sparks29032** [comment]
pre-commit.ci run

**Comment by Sparks29032** [comment]
From now on each new PR should automatically run this pre-commit once merged. Failures should be fixed with Caden's PR.

**Comment by Sparks29032** [comment]
> I linked diffpy org at pre-commit.ci. I wasn't quite sure what else I needed to do?

Everything should be set up and good to go. We can test this after merging this PR and putting in a new PR to see if it runs this CI automatically.

**Comment by sbillinge** [comment]
super, thanks @Sparks29032

**Comment by sbillinge** [comment]
@Sparks29032 @cadenmyers13 I tested this locally and it seemed to work (no commit to main after updating merged main).

So @cadenmyers13 I think we need to migrate these changes also to diffpy.utils and diffpy.labpdfproc.

Thanks so much @Sparks29032 for this!

**Pull request merged**